plugins/azure/resoto_plugin_azure/resource/metrics.py

```diff
@@ -0,0 +1,233 @@
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("id"),
+        "type": S("type"),
+        "name": S("name") >> Bend(AzureMetricValueName.mapping),
+        "displayDescription": S("displayDescription"),
+        "unit": S("unit"),
+        "timeseries": S("timeseries", default=[]) >> ForallBend(AzureMetricTimeSeries.mapping),
+    }
+    id: Optional[str] = field(default=None)
+    type: Optional[str] = field(default=None)
+    name: Optional[str] = field(default=None)
+    display_description: Optional[str] = field(default=None)
+    unit: Optional[str] = field(default=None)
+    timeseries: Optional[List[AzureMetricTimeSeries]] = field(default=None)
+
+
+@define(hash=True, frozen=True)
+class AzureMetricQuery:
+    metric_name: str
+    metric_namespace: str
+    ref_id: str
+    instance_id: str
+    metric_id: str
+    aggregation: str = "total"
+    unit: str = "Count"
+
+    @staticmethod
+    def create(
+        metric_name: str,
+        metric_namespace: str,
+        instance_id: str,
+        ref_id: str,
+        metric_id: Optional[str] = None,
+        aggregation: str = "total",
+        unit: str = "Count",
+    ) -> "AzureMetricQuery":
+        metric_id = f"{instance_id}/providers/Microsoft.Insights/metrics/{metric_name}"
+        # noinspection PyTypeChecker
+        return AzureMetricQuery(
+            metric_name=metric_name,
+            metric_namespace=metric_namespace,
+            instance_id=instance_id,
+            metric_id=metric_id,
+            aggregation=aggregation,
+            ref_id=ref_id,
+            unit=unit,
+        )
+
+
+@define(eq=False, slots=False)
+class AzureMetricData:
+    kind: ClassVar[str] = "azure_metric"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "timespan": S("timespan"),
+        "interval": S("interval"),
+        "namespace": S("namespace"),
+        "resource_region": S("resourceregion"),
+        "full_metric_values_data": S("value") >> ForallBend(AzureMetricValue.mapping),
+    }
+    full_metric_values_data: List[AzureMetricValue] = field(factory=list)
+    metric_id: Optional[str] = field(default=None)
+    metric_values: Optional[List[float]] = field(default=None)
+    metric_timestamps: Optional[List[datetime]] = field(default=None)
+    timespan: Optional[str] = field(default=None)
+    interval: Optional[str] = field(default=None)
+    namespace: Optional[str] = field(default=None)
+    resource_region: Optional[str] = field(default=None)
+
+    def set_values(self, query_aggregation: str) -> None:
+        if self.full_metric_values_data:
+            metric_values_result = [
+                data
+                for metric_value in self.full_metric_values_data
+                for timeseries in metric_value.timeseries or []
+                for data in timeseries.data or []
+            ]
+            self.metric_values = [getattr(metric, query_aggregation) for metric in metric_values_result][::-1]
```

Comment:
I try to understand this line. Several thoughts:
- do we only use the last element?
- if yes why are we getting multiple elements? 
- there needs to be a comment in the code that explains this

Reply:
We do not take only last element. We reverse the list from last to first element.